src/backend/TrafficCourts/Staff.Service/Services/DisputeService.cs

```
@@ -241,21 +241,32 @@ public async Task SubmitDisputeAsync(long disputeId, ClaimsPrincipal user, Cance
         }
- 
-        // Status to PROCESSING
-        dispute = await _oracleDataApi.SubmitDisputeAsync(disputeId, cancellationToken);
 
-        // TCVP-2977: Get disputant name data from ICBC RSI ticket search and update dispute's IcbcNameDetail
-        GetDisputeOptions options = new() {  DisputeId = disputeId, Assign = false, GetNameFromIcbc = true };
-        await GetIcbcTicketInformation(dispute, options, cancellationToken);
+        try
+        {
+            // These processes are necessary for the correct update and submit to ARC, but they can fail (for example if the RSI API is down). 
+            // If they fail, we need to leave the Dispute Status as NEW, otherwise it'll be PROCESSING, but not actually submitted to ARC... and then it will sit there forever stalled.
+            // TCVP-3361 for more detail
 
-        // set AddressProvince to 2 character abbreviation code if prov seq no & ctry id present
-        if (dispute.AddressProvinceSeqNo is not null && dispute.AddressCountryId is not null)
+            // TCVP-2977: Get disputant name data from ICBC RSI ticket search and update dispute's IcbcNameDetail
+            GetDisputeOptions options = new() { DisputeId = disputeId, Assign = false, GetNameFromIcbc = true };
+            await GetIcbcTicketInformation(dispute, options, cancellationToken);
+
+            // set AddressProvince to 2 character abbreviation code if prov seq no & ctry id present
+            if (dispute.AddressProvinceSeqNo is not null && dispute.AddressCountryId is not null)
+            {
+                var provFound = await _provinceLookupService.GetByProvSeqNoCtryIdAsync(dispute.AddressProvinceSeqNo.Value, dispute.AddressCountryId.Value, cancellationToken);
+                if (provFound != null) { dispute.AddressProvince = provFound.ProvAbbreviationCd; }
+            }
+        }
+        catch (Exception)

```

` GetIcbcTicketInformation()` internally catches and swallows all exceptions (it only logs them).
I think this means any failure inside `GetIcbcTicketInformation()` will NOT trigger this outer catch block, and execution will continue normally

---

We were getting incorrect behavior when RSI calls were timing out. It would still crash somehow instead of just eating and logging the exceptions. Unfortunately it's hard to test. Disabling the endpoint gives a 404 or whatever, which does what you said - keeps going. The timeouts somehow bypass the try - because they're framework level I think? With the code re-ordered like this, if that exception happens, it doesn't change the dispute to processing, and the staff know to try again.